docs/explanations/how-workflows-work.md

```diff
@@ -10,11 +10,12 @@ In this guide, we'll explain how DBOS makes workflows reliable.
 
 ### Workflow Guarantees
 
-To reiterate from our [workflow tutorial](../tutorials/workflow-tutorial), workflows provide the following reliability guaranteees:
+To reiterate from our [workflow tutorial](../tutorials/workflow-tutorial), workflows provide the following reliability guaranteees.
+These guarantees assume that the application and database may crash and go offline at any point in time, but are always restarted and return online.
 
-1.  They always run to completion.  If a server executing a workflow crashes and restarts, it resumes all incomplete workflows.
-2.  Transactions execute _exactly once_.  Regardless of what failures occur during a workflow's execution, it executes each of its transactions once and only once.
-3.  Communicators execute _at least once_ but are never re-executed after they successfully complete.  If a failure occurs inside a communicator, the communicator may be retried, but once a communicator has completed execution, DBOS guarantees it will never be re-executed regardless of what failures happen afterwards.
+1.  Workflows always run to completion.  If a DBOS process crashes and is restarted while executing a workflow, it resumes the workflow from where it left off.
+2.  Transactions execute _exactly once_.  Once a workflow commits a transaction, it will never re-execute that transaction.
```

Comment:
If we want to be more precise, we can say transactions *commit* exactly once. And maybe explain what happens if a transaction fails.